src/components/logo/logo.tsx

```diff
@@ -27,14 +30,60 @@ const mapStateToProps = state => ({
  */
 @connect(mapStateToProps)
 @withPlayer
+@withEventManager
 @withLogger(COMPONENT_NAME)
 @withText({logoText: 'controls.logo'})
 class Logo extends Component<any, any> {
+  private _logoRef: RefObject<HTMLAnchorElement> = createRef<HTMLAnchorElement>();
+
+  /**
+   * before component mounted, set initial state of the logo
+   *
+   * @returns {void}
+   * @memberof Logo
+   */
+  public componentWillMount(): void {
+    this.setState({isUrlClickable: true});
+  }
+
+  /**
+   * when component did mount
+   *
+   * @returns {void}
+   * @memberof Logo
+   */
+  public componentDidMount(): void {
+    this._handleLogoUrl();
+  }
+
+  /**
+   * handles the logo url
+   * if the url contains ${entryId}, then replace it with the played entry id
+   *
+   * @returns {void}
+   * @memberof Logo
+   */
+  private _handleLogoUrl(): void {
+    const url = this.props.config.url;
+    if (url && url.indexOf(ENTRY_VAR) !== -1) {
+      const {player, eventManager} = this.props;
+      eventManager.listen(player, player.Event.MEDIA_LOADED, () => {
+        const entryId = player.sources.id;
+        if (this._logoRef.current && typeof entryId === 'string') {
+          this._logoRef.current.href = url.replace(ENTRY_VAR, entryId);
```

Comment:
Why do you need accessing via ref and not just keeping url statse in the component???
It is the correct way how the component changes should be kept  and easier for debugging